archive/t/typescript/roman-numeral.ts

```diff
@@ -0,0 +1,75 @@
+// Roman Numeral Conversion
+
+const romanNumeralValues = {
+  "I": 1,
+  "V": 5,
+  "X": 10,
+  "L": 50,
+  "C": 100,
+  "D": 500,
+  "M": 1000
+};
+
+function romanNumeralConversion(romanNumeral?: string) {
+
+  // Check for blank input
+  if (romanNumeral === undefined) {
+    return "Usage: please provide a string of roman numerals";
+  }
+
+  // Check for empty string
+  if (romanNumeral === "") {
+    return 0;
+  }
+
+  // Split the string
+  const romans = romanNumeral.split("");
+  // Get valid characters
+  const validValues = Object.keys(romanNumeralValues)
+
+  // Check for invalid characters
+  for (const roman of romans) {
+    if (roman === undefined || validValues.indexOf(roman) === -1) {
+      return "Error: " + roman + " is not a valid string of roman numerals";
```

Comment:
It looks like there's still one more [failure](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6397954278/job/17367161399?pr=3407#step:6:7643).